src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentDescriptorProvider.cs

```diff
@@ -14,6 +16,8 @@ namespace Microsoft.AspNet.Mvc.ViewComponents
     /// </summary>
     public class DefaultViewComponentDescriptorProvider : IViewComponentDescriptorProvider
     {
+        private const string AsyncMethodName = "InvokeAsync";
+        private const string SyncMethodName = "Invoke";
```

Comment:
So it's always `@Component.InvokeAsync` but you can write your `ViewComponent` with `Invoke` OR `InvokeAsync`? Interesting.